toolkit/tools/pkgworker/pkgworker.go

```diff
@@ -264,63 +252,102 @@ func installBuildRequires(defines map[string]string) (err error) {
 		return
 	}
 
-	if len(buildRequires) > 0 {
-		defaultArgs := []string{"install", "-y"}
-		installArgs := make([]string, 0, len(buildRequires)+len(defaultArgs))
+	if len(buildRequires) == 0 {
+		return
+	}
 
-		installArgs = append(installArgs, defaultArgs...)
+	packagesToInstall := make([]string, 0, len(buildRequires))
+	for _, pkg := range buildRequires {
+		// Replace version conditionals with tdnf friendly version:
+		// - replace >= with "latest" (no version)
+		// - replace = and <= with the exact version
+		buildReq := greaterThanOrEqualRegex.ReplaceAllString(pkg, "")
+		buildReq = equalToRegex.ReplaceAllString(buildReq, "-")
+		buildReq = lessThanOrEqualToRegex.ReplaceAllString(buildReq, "-")
+
+		// Add each package to the installArgs
+		packagesToInstall = append(packagesToInstall, strings.TrimSpace(buildReq))
+	}
 
-		for _, pkg := range buildRequires {
-			// Replace version conditionals with tdnf friendly version:
-			// - replace >= with "latest" (no version)
-			// - replace = and <= with the exact version
-			buildReq := greaterThanOrEqualRegex.ReplaceAllString(pkg, "")
-			buildReq = equalToRegex.ReplaceAllString(buildReq, "-")
-			buildReq = lessThanOrEqualToRegex.ReplaceAllString(buildReq, "-")
+	// Try matching dependencies from locally build RPMs first. If some are not available,
+	// then consider the package cache. This will ensure that local packages are preferred over remote ones.
+	failedToInstall, err := tdnfInstall(packagesToInstall, allowCacheInitially)
```

Comment:
>allowCacheInitially [](start = 56, length = 19)

This is just confusing with the current name. I was convinced that we actually are allowing the cache here. Considering we might be calling this function twice, I'd suggest defining a variable `allowCache := false` right above this line and then flipping it to `true` if the installation fails (so pretty much what you're already doin, just without the `const`).

Reply:
Thanks for the update but it looks like now we're starting with allowing the cache and then disabling that, so I think the opposite of what we were going for initially.

---
In reply to: [500638104](https://github.com/microsoft/CBL-Mariner/pull/141#discussion_r500638104) [](ancestors = 500638104)